File: nodejs/base/api/lib/routing/defaults/default-route-collection.ts
import { McmaResource } from "@mcma/core";
import { DefaultRouteBuilderType } from "./default-route-builder-type";
import { DefaultRoutes } from "./default-routes";
import { DefaultRouteBuilder } from "./default-route-builder";

export class DefaultRouteCollection<T extends McmaResource> {
    included: DefaultRouteBuilderType[] = [];
    query: DefaultRouteBuilder<T[]>;
    create: DefaultRouteBuilder<T>;
    get: DefaultRouteBuilder<T>;
    update: DefaultRouteBuilder<T>;
    delete: DefaultRouteBuilder<T>;

    constructor(initialRoutes: DefaultRoutes<T>) {
        this.query = initialRoutes?.query;
        this.create = initialRoutes?.create;
        this.get = initialRoutes?.get;
        this.update = initialRoutes?.update;
        this.delete = initialRoutes?.delete;
    }
    
    addAll(): this {
        this.included.push(this.query, this.create, this.get, this.update, this.delete);
        return this;
    }
}

File: dotnet/base/Mcma.Api/Routing/Defaults/DefaultRouteCollection.cs
using System.Collections.Generic;
using Mcma;

namespace Mcma.Api.Routing.Defaults
{
    public class DefaultRouteCollection<TResource> where TResource : McmaResource
    {           
        internal DefaultRouteCollection()
        {
        }

        internal List<IDefaultRouteBuilder> Included { get; } = new List<IDefaultRouteBuilder>();

        internal void AddAll() => Included.AddRange(new IDefaultRouteBuilder[] { Query, Create, Get, Update, Delete });

        public DefaultRouteBuilder<IEnumerable<TResource>> Query { get; internal set; }

        public DefaultRouteBuilder<TResource> Create { get; internal set; }

        public DefaultRouteBuilder<TResource> Get { get; internal set; }

        public DefaultRouteBuilder<TResource> Update { get; internal set; }

        public DefaultRouteBuilder<TResource> Delete { get; internal set; }
    }
}
